server/src/main/java/io/crate/protocols/postgres/Cursor.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.protocols.postgres;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+import javax.annotation.Nullable;
+
+import io.crate.action.sql.PreparedStmt;
+import io.crate.action.sql.ResultReceiver;
+import io.crate.action.sql.RowConsumerToResultReceiver;
+import io.crate.analyze.AnalyzedStatement;
+import io.crate.data.BatchIterator;
+import io.crate.data.Row;
+
+/**
+ * Cursor is created by Declare and reused throughout successive Fetches then removed when Close is called.
+ */
+public class Cursor extends Portal {
+
+    public enum State {
+        Declare, Fetch
+    }
+
+    private PreparedStmt preparedStmt; // hides Portal.preparedStmt
```

Comment:
why can't we reuse the super class `preparedStmt`?

Reply:
Thanks, It is better to reuse `Portal.preparedStmt`.